lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
It is quite a big improvement because it ensures this will work with different rails versions. What if rails changes format of that object or that method? Because it is a mock no one can know because it does not test the rails interface. 

The tests depend on rails. They include methods from ActionView, ActiveModel and other parts. Using object from ActiveRecord and include few methods is not going to include more dependencies.